From 188662a7522cb4b3e2d07fe73d81ba92a39f8caa Mon Sep 17 00:00:00 2001
From: Josua Mayer <josua@solid-run.com>
Date: Thu, 19 Jun 2025 13:22:43 +0300
Subject: [PATCH] package/freescale-imx/imx-vpu-hantro: hack enable build w/o
 kernel

Allow building imx-vpu-hantro packages without BR2_KERNEL set.
This is useful when the kernel is being built separately,
as part of the SolidRun BSP.

This is considered a hack in that it avoids using buildroot correctly
to handle kernel compilation, and as it hardcodes the path to the
external linux sources.

Signed-off-by: Josua Mayer <josua@solid-run.com>
---
 package/freescale-imx/imx-vpu-hantro-daemon/Config.in         | 1 -
 .../imx-vpu-hantro-daemon/imx-vpu-hantro-daemon.mk            | 4 ++--
 package/freescale-imx/imx-vpu-hantro-vc/Config.in             | 1 -
 package/freescale-imx/imx-vpu-hantro-vc/imx-vpu-hantro-vc.mk  | 1 -
 package/freescale-imx/imx-vpu-hantro/Config.in                | 1 -
 package/freescale-imx/imx-vpu-hantro/imx-vpu-hantro.mk        | 3 +--
 6 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/package/freescale-imx/imx-vpu-hantro-daemon/Config.in b/package/freescale-imx/imx-vpu-hantro-daemon/Config.in
index 663ed434..0829badc 100644
--- a/package/freescale-imx/imx-vpu-hantro-daemon/Config.in
+++ b/package/freescale-imx/imx-vpu-hantro-daemon/Config.in
@@ -7,7 +7,6 @@ comment "imx-vpu-hantro-daemon needs an i.MX platform with Hantro VPU"
 
 config BR2_PACKAGE_IMX_VPU_HANTRO_DAEMON
 	bool "imx-vpu-hantro-daemon"
-	depends on BR2_LINUX_KERNEL
 	depends on BR2_PACKAGE_FREESCALE_IMX_HAS_VPU_HANTRO
 	select BR2_PACKAGE_IMX_VPU_HANTRO
 	select BR2_PACKAGE_IMX_VPU_HANTRO_VC if BR2_PACKAGE_FREESCALE_IMX_PLATFORM_IMX8MP
diff --git a/package/freescale-imx/imx-vpu-hantro-daemon/imx-vpu-hantro-daemon.mk b/package/freescale-imx/imx-vpu-hantro-daemon/imx-vpu-hantro-daemon.mk
index ab4746be..b1c57f40 100644
--- a/package/freescale-imx/imx-vpu-hantro-daemon/imx-vpu-hantro-daemon.mk
+++ b/package/freescale-imx/imx-vpu-hantro-daemon/imx-vpu-hantro-daemon.mk
@@ -9,7 +9,7 @@ IMX_VPU_HANTRO_DAEMON_SITE = $(FREESCALE_IMX_SITE)
 IMX_VPU_HANTRO_DAEMON_LICENSE = NXP Semiconductor Software License Agreement
 IMX_VPU_HANTRO_DAEMON_LICENSE_FILES = LICENSE.txt
 IMX_VPU_HANTRO_DAEMON_REDISTRIBUTE = NO
-IMX_VPU_HANTRO_DAEMON_DEPENDENCIES = imx-vpu-hantro linux
+IMX_VPU_HANTRO_DAEMON_DEPENDENCIES = imx-vpu-hantro
 ifeq ($(BR2_PACKAGE_FREESCALE_IMX_PLATFORM_IMX8MP),y)
 IMX_VPU_HANTRO_DAEMON_DEPENDENCIES += imx-vpu-hantro-vc
 endif
@@ -18,7 +18,7 @@ IMX_VPU_HANTRO_DAEMON_MAKE_ENV = \
 	$(TARGET_MAKE_ENV) \
 	CROSS_COMPILE="$(TARGET_CROSS)" \
 	SDKTARGETSYSROOT=$(STAGING_DIR) \
-	LINUX_KERNEL_ROOT=$(LINUX_DIR) \
+	LINUX_KERNEL_ROOT=$(TOPDIR)/../linux-imx \
 	CTRLSW_HDRPATH="$(STAGING_DIR)/usr/include" \
 	PLATFORM=$(BR2_PACKAGE_FREESCALE_IMX_PLATFORM)
 
diff --git a/package/freescale-imx/imx-vpu-hantro-vc/Config.in b/package/freescale-imx/imx-vpu-hantro-vc/Config.in
index 9a0f448b..3d7cddc5 100644
--- a/package/freescale-imx/imx-vpu-hantro-vc/Config.in
+++ b/package/freescale-imx/imx-vpu-hantro-vc/Config.in
@@ -7,7 +7,6 @@ comment "imx-vpu-hantro-vc needs an i.MX platform with Hantro VPU"
 
 config BR2_PACKAGE_IMX_VPU_HANTRO_VC
 	bool "imx-vpu-hantro-vc"
-	depends on BR2_LINUX_KERNEL
 	depends on BR2_PACKAGE_FREESCALE_IMX_HAS_VPU_HANTRO
 	help
 	  i.MX VC8000E Encoder library. Intended for the i.MX 8M Plus
diff --git a/package/freescale-imx/imx-vpu-hantro-vc/imx-vpu-hantro-vc.mk b/package/freescale-imx/imx-vpu-hantro-vc/imx-vpu-hantro-vc.mk
index 3d266fb5..cb725527 100644
--- a/package/freescale-imx/imx-vpu-hantro-vc/imx-vpu-hantro-vc.mk
+++ b/package/freescale-imx/imx-vpu-hantro-vc/imx-vpu-hantro-vc.mk
@@ -7,7 +7,6 @@
 IMX_VPU_HANTRO_VC_VERSION = 1.9.0
 IMX_VPU_HANTRO_VC_SITE = $(FREESCALE_IMX_SITE)
 IMX_VPU_HANTRO_VC_SOURCE = imx-vpu-hantro-vc-$(IMX_VPU_HANTRO_VC_VERSION).bin
-IMX_VPU_HANTRO_VC_DEPENDENCIES = linux
 IMX_VPU_HANTRO_VC_INSTALL_STAGING = YES
 
 IMX_VPU_HANTRO_VC_LICENSE = NXP Semiconductor Software License Agreement
diff --git a/package/freescale-imx/imx-vpu-hantro/Config.in b/package/freescale-imx/imx-vpu-hantro/Config.in
index 25ce50ff..204a0a0f 100644
--- a/package/freescale-imx/imx-vpu-hantro/Config.in
+++ b/package/freescale-imx/imx-vpu-hantro/Config.in
@@ -7,7 +7,6 @@ comment "imx-vpu-hantro needs an i.MX platform with Hantro VPU"
 
 config BR2_PACKAGE_IMX_VPU_HANTRO
 	bool "imx-vpu-hantro"
-	depends on BR2_LINUX_KERNEL
 	depends on BR2_PACKAGE_FREESCALE_IMX_HAS_VPU_HANTRO
 	help
 	  Library of userspace helpers specific for the NXP i.MX SoC
diff --git a/package/freescale-imx/imx-vpu-hantro/imx-vpu-hantro.mk b/package/freescale-imx/imx-vpu-hantro/imx-vpu-hantro.mk
index 0987551c..76680e57 100644
--- a/package/freescale-imx/imx-vpu-hantro/imx-vpu-hantro.mk
+++ b/package/freescale-imx/imx-vpu-hantro/imx-vpu-hantro.mk
@@ -7,7 +7,6 @@
 IMX_VPU_HANTRO_VERSION = 1.29.0
 IMX_VPU_HANTRO_SITE = $(FREESCALE_IMX_SITE)
 IMX_VPU_HANTRO_SOURCE = imx-vpu-hantro-$(IMX_VPU_HANTRO_VERSION).bin
-IMX_VPU_HANTRO_DEPENDENCIES = linux
 IMX_VPU_HANTRO_INSTALL_STAGING = YES
 
 IMX_VPU_HANTRO_MAKE_ENV = \
@@ -15,7 +14,7 @@ IMX_VPU_HANTRO_MAKE_ENV = \
 	$(TARGET_CONFIGURE_OPTS) \
 	CROSS_COMPILE="$(TARGET_CROSS)" \
 	SDKTARGETSYSROOT=$(STAGING_DIR) \
-	LINUX_KERNEL_ROOT=$(LINUX_DIR) \
+	LINUX_KERNEL_ROOT=$(TOPDIR)/../linux-imx \
 	PLATFORM=$(BR2_PACKAGE_FREESCALE_IMX_PLATFORM)
 
 IMX_VPU_HANTRO_LICENSE = NXP Semiconductor Software License Agreement
-- 
2.43.0

